dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/event/RumEventMapper.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain.event
+
+import com.datadog.android.core.internal.event.NoOpEventMapper
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.event.EventMapper
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.model.ViewEvent
+
+internal data class RumEventMapper(
+    val viewEventMapper: EventMapper<ViewEvent> = NoOpEventMapper(),
+    val errorEventMapper: EventMapper<ErrorEvent> = NoOpEventMapper(),
+    val resourceEventMapper: EventMapper<ResourceEvent> = NoOpEventMapper(),
+    val actionEventMapper: EventMapper<ActionEvent> = NoOpEventMapper()
+) : EventMapper<RumEvent> {
+
+    override fun map(event: RumEvent): RumEvent? {
+        val bundledMappedEvent = when (val bundledEvent = event.event) {
+            is ViewEvent -> viewEventMapper.map(bundledEvent)
+            is ActionEvent -> actionEventMapper.map(bundledEvent)
+            is ErrorEvent -> errorEventMapper.map(bundledEvent)
+            is ResourceEvent -> resourceEventMapper.map(bundledEvent)
+            else -> {
+                sdkLogger.w(
+                    "RumEventMapper: there was no EventMapper assigned for" +
+                        " RUM event type: [${bundledEvent.javaClass.simpleName}]"
+                )
+                bundledEvent
```

Comment:
Isn't this a case when one of the public `setRum###EventMapper()` APIs was not used? IMO we shouldn't print a warning in such case, as those are opt-in APIs without a mandatory use.

Reply:
Nope. If there's no EventMapper used we will assign one by default : see `NoOpEventMapper`.

Reply:
This is when there's actually a new event type for which we did not provide a NoOpEventMapper meaning that we forgot to provide a public API method with a real implementation of the EventMapper<T>

Reply:
I see 👌 thanks for explaining 🙂